diff --git a/api/src/services/versions.ts b/api/src/services/versions.ts
index 72765ca..a80373c 100644
--- a/api/src/services/versions.ts
+++ b/api/src/services/versions.ts
@@ -226,40 +226,46 @@ export class VersionsService extends ItemsService<ContentVersion> {
 			schema: this.schema,
 		});
 
-		const activityService = new ActivityService({
-			knex: this.knex,
-			schema: this.schema,
-		});
-
-		const revisionsService = new RevisionsService({
-			knex: this.knex,
-			schema: this.schema,
-		});
-
 		const { item, collection, delta: existingDelta } = version;
 
-		const activity = await activityService.createOne({
-			action: Action.VERSION_SAVE,
-			user: this.accountability?.user ?? null,
-			collection,
-			ip: this.accountability?.ip ?? null,
-			user_agent: this.accountability?.userAgent ?? null,
-			origin: this.accountability?.origin ?? null,
-			item,
-		});
-
 		const helpers = getHelpers(this.knex);
 
 		let revisionDelta = await payloadService.prepareDelta(delta);
 
-		await revisionsService.createOne({
-			activity,
-			version: key,
-			collection,
-			item,
-			data: revisionDelta,
-			delta: revisionDelta,
-		});
+		const trackingAccountability = this.schema.collections[collection]?.accountability ?? null;
+
+		if (trackingAccountability !== null) {
+			const activityService = new ActivityService({
+				knex: this.knex,
+				schema: this.schema,
+			});
+
+			const activity = await activityService.createOne({
+				action: Action.VERSION_SAVE,
+				user: this.accountability?.user ?? null,
+				collection,
+				ip: this.accountability?.ip ?? null,
+				user_agent: this.accountability?.userAgent ?? null,
+				origin: this.accountability?.origin ?? null,
+				item,
+			});
+
+			if (trackingAccountability === 'all') {
+				const revisionsService = new RevisionsService({
+					knex: this.knex,
+					schema: this.schema,
+				});
+
+				await revisionsService.createOne({
+					activity,
+					version: key,
+					collection,
+					item,
+					data: revisionDelta,
+					delta: revisionDelta,
+				});
+			}
+		}
 
 		revisionDelta = revisionDelta ? revisionDelta : null;
 
